src/test/scala/com/tsunderebug/scolortesting/otf/types/SequentialMapGroupTest.scala

```diff
@@ -0,0 +1,42 @@
+package com.tsunderebug.scolor.otf.types
+
+import org.scalatest.{FlatSpec, OptionValues}
+import spire.math.UInt
+
+import com.tsunderebug.scolor.otf.{OTFByteAllocator, OpenTypeFont}
+
+class SequentialMapGroupTest() extends FlatSpec with OptionValues {
+
+  private def byteAllocator = new OTFByteAllocator(OpenTypeFont(Nil))
+
+  "SequentialMapGroup" should "be sorted naturally by their starting character code" in {
+      val range = (0 to 10 by 2)
+      val incorrectlyOrdered = for (i <- range.reverse) yield SequentialMapGroup(UInt(i), UInt(i +1), UInt(0))
+      val correctlyOrdered = for (i <- range) yield SequentialMapGroup(UInt(i), UInt(i +1), UInt(0))
+      assertResult(correctlyOrdered) {
+        incorrectlyOrdered.sorted
+      }
+    }
+
+    it should "have a length of 12" in {
```

Comment:
I could probably expand on this, but I would need to know _why_ it is 12 right now. Is it something about having 3 data sections (with each being 4 bytes?)

Reply:
All length values are in bytes :)